src/main/java/com/rewedigital/composer/composing/IncludedService.java

```diff
@@ -4,24 +4,34 @@
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.spotify.apollo.Response;
 
 class IncludedService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(IncludedService.class);
+
     public static class WithResponse {
         private final int startOffset;
         private final int endOffset;
         private final Response<String> response;
+        private final CompositionStep step;
 
-        private WithResponse(final int startOffset, final int endOffset, final Response<String> response) {
+        private WithResponse(final CompositionStep step, final int startOffset, final int endOffset,
+            final Response<String> response) {
             this.startOffset = startOffset;
             this.endOffset = endOffset;
             this.response = response;
+            this.step = step;
+
+            LOGGER.debug("included service response: {} received via {}", response, step.callStack());

```

Comment:
Could we add guard with LOGGER.isDebugEnabled as the call to `callStack` has minor but cost assigned.